### ISSUE TITLE
Missing description of kernel_bundle::get_kernel_info

### ISSUE THREAD

**Issue by dm-vodopyanov** [KhronosGroup/SYCL-Docs#376]
According to SYCL 2020 Rev 6:

> Using a group function inside of a kernel may introduce additional limits on the resources available to user code inside the same kernel. The behavior of these limits is implementation-defined, but must be reflected by calls to kernel querying functions (such as kernel_bundle::get_kernel_info) as described in [Section 4.11.12.1](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:bundles.query).

`kernel_bundle::get_kernel_info` mentioned only once in the spec in this quote above.

**Comment by dm-vodopyanov** [comment]
And could you please clarify the meaning of

> but must be reflected by calls to kernel querying functions

?

**Comment by gmlueck** [comment]
I think the name `kernel_bundle::get_kernel_info` is a typo.  This should say `kernel::get_info` instead.  The section cross-reference should also be updated to [4.11.13.1](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_queries).

Maybe @Pennycook has some ideas on clarifying which resources could be limited by calling the group functions.  I can imagine that some group functions may use SLM, so this would reduce the amount of SLM available to user code in the kernel.  However, there is no `kernel::get_info` query that can tell you the amount of SLM used by a kernel, so this is not a good example.  (See #188, however, for a request to add this query.)

**Comment by gmlueck** [comment]
I fixed the typo in #377.  We may want another PR to clarify the wording about which resources might be limited.

**Comment by Pennycook** [comment]
> Maybe @Pennycook has some ideas on clarifying which resources could be limited by calling the group functions. I can imagine that some group functions may use SLM, so this would reduce the amount of SLM available to user code in the kernel. However, there is no kernel::get_info query that can tell you the amount of SLM used by a kernel, so this is not a good example. (See https://github.com/KhronosGroup/SYCL-Docs/issues/188, however, for a request to add this query.)

Yeah, this is exactly what we had in mind.  A lot of the group algorithms wording was derived from our [original extension](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/deprecated/sycl_ext_oneapi_group_algorithms.asciidoc), which is slightly more explicit here:

> "Using functions from the group algorithms library inside of a kernel may introduce additional limits on the resources available to user code inside the same kernel (e.g. private memory, work-group local memory). The behavior of these limits is implementation-defined, but must be reflected by calls to kernel querying functions such as `kernel::get_info`."

(It looks like at some point we fixed the typo in the extension, and forgot to check the SYCL specification.)

I think we were assuming that this [other extension](https://github.com/intel/llvm/tree/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_local_static_mem_used.asciidoc) would also make it into SYCL 2020, but it didn't.

However, looking at this again with fresh eyes, I can see that querying the amount of _static_ local memory used by a kernel is insufficient anyway.  I'll add a note to #188.

**Comment by gmlueck** [comment]
Shall we close this issue them when #377 is merged?  We can clarify the wording about which resources are limited whenever we add an API to query SLM usage.
